Formula/tagref.rb

```diff
@@ -0,0 +1,45 @@
+class Tagref < Formula
+  desc "Refer to other locations in your codebase"
+  homepage "https://github.com/stepchowfun/tagref"
+  url "https://github.com/stepchowfun/tagref/archive/v1.5.0.tar.gz"
+  sha256 "dd6321133c2bef64f9230d6aaddfba8a4327749236638c23c65d0832ca2fef48"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    (testpath/"file-1.txt").write <<~EOS
+      Here's a reference to the tag below: [ref:foo]
+      Here's a reference to a tag in another file: [ref:bar]
+      Here's a tag: [tag:foo]
+    EOS
+
+    (testpath/"file-2.txt").write <<~EOS
+      Here's a tag: [tag:bar]
+    EOS
+
+    output, status = Open3.capture2e({ "NO_COLOR" => "true" }, "#{bin}/tagref")
+    assert(status.success?, "Tagref did not exit successfully.")
```

Comment:
```suggestion
    with_env(NO_COLOR: true) do
      output = shell_output("#{bin}/tagref 2>&1")
    end
```
Isn't this the same?

Reply:
Applied your suggestion—thanks!

Reply:
Same below, please.

Reply:
I had to revert this change because it isn't the same. The `output` from `Open3.capture2e` includes STDERR, but the result of `shell_output` does not seem to include that. (Tagref outputs the message to STDERR.)

Reply:
Isn't that what `2>&1` does?

Reply:
Oh, yeah—you're right. Hm. But for some reason the change to `shell_output` causes the test to fail with this error:

```
$ brew test tagref
==> Testing tagref
Warning: Failed setting group "admin" on /private/tmp/tagref-test-20220207-17181-1xzhlvf
Error: tagref: failed
An exception occurred within a child process:
  TypeError: no implicit conversion of true into String
/usr/local/Homebrew/Library/Homebrew/utils.rb:585:in `[]='
/usr/local/Homebrew/Library/Homebrew/utils.rb:585:in `block in with_env'
/usr/local/Homebrew/Library/Homebrew/utils.rb:582:in `each'
/usr/local/Homebrew/Library/Homebrew/utils.rb:582:in `with_env'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/tagref.rb:25:in `block in <class:Tagref>'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2109:in `block (3 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/utils.rb:588:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2108:in `block (2 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/formula.rb:948:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2107:in `block in run_test'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `block in run'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `chdir'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `run'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2360:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2101:in `run_test'
/usr/local/Homebrew/Library/Homebrew/test.rb:43:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:93:in `block in timeout'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `block in catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/timeout.rb:108:in `timeout'
/usr/local/Homebrew/Library/Homebrew/test.rb:48:in `<main>'
```

Do you have any ideas as to why that is happening? Otherwise do we need this change?

Reply:
Aha. I figured it out. It needs to be:

```ruby
with_env(NO_COLOR: "true") do
```

instead of

```ruby
with_env(NO_COLOR: true) do
```

And also the `output` variable needs to be introduced outside the scope of the block:

```ruby
    output = nil
    with_env(NO_COLOR: 'true') do
      output = shell_output("#{bin}/tagref 2>&1")
    end
```

Otherwise, a different error emerges: `NameError: undefined local variable or method `output'`

Personally, I don't find this to be an improvement over what I had. But I don't have strong feelings about it, so I'm happy to make the change. Done.

The one below actually tests that the command fails, so I don't think it can be converted into this style (but correct me if I'm mistaken).